pkg/scale/encode.go

```diff
@@ -0,0 +1,420 @@
+package scale
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"reflect"
+)
+
+func Marshal(v interface{}) (b []byte, err error) {
+	es := encodeState{
+		fieldScaleIndicesCache: cache,
```

Comment:
is this needed? since `cache` is a package variable

Reply:
I like having localized struct attributes.  May make it easier to test.